### ISSUE TITLE
[QUESTION] udemy-dl does'nt seem to isntall even after successfully installed message

### ISSUE THREAD

**Issue by monodeepdas1215** [nishad/udemy-dl#265]
Please prefix your issue with one of the following: [BUG], [PROPOSAL], [QUESTION].

Replace everything between stars with current version of your udemy-dl (output of udemy-dl -v):  
Version : **X.X.X**

**In raising this issue, I confirm the following (please check boxes, eg [X]):**

- [ ] I have read and understood the [contributors guide](https://github.com/nishad/udemy-dl/blob/master/.github/CONTRIBUTING.md).
- [ ] I have checked that the bug-fix I am reporting can be replicated, or that the feature I am suggesting isn't already present.
- [X] I have checked that the issue I'm posting isn't already reported.
- [ ] I have checked that the i
![screenshot-2017-10-27 screenshot from 2017-10-27 12-50-37 png png image 1920 x 1080 pixels - scaled 87](https://user-images.githubusercontent.com/15053941/32092359-bee2ce9e-bb15-11e7-880b-800963203ed7.png)
ssue I'm porting isn't already solved and no duplicates exist in [closed issues](https://github.com/nishad/udemy-dl/issues?q=is%3Aissue+is%3Aclosed) and in [opened issues](https://github.com/nishad/udemy-dl/issues)
- [ ] I have checked the pull requests tab for existing solutions/implementations to my issue/suggestion.

**My familiarity with the project is as follows (check one, eg [X]):**

- [ ] I have never used the project.
- [X ] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

**If the project maintainer has any additional requirements, they will be listed here.**

- No additional requirements.

---

{issue content here, replace this line with your issue content}

**Comment by varunchandak** [comment]
tell the output of the following commands

`which udemy-dl`
whereis udemy-dl`
`echo $PATH`

**Comment by monodeepdas1215** [comment]
![screenshot-2017-10-28 screenshot from 2017-10-28 14-17-45 png png image 1920 x 1080 pixels - scaled 87](https://user-images.githubusercontent.com/15053941/32132937-ff692d84-bbea-11e7-8881-c652dce6e186.png)

**Comment by varunchandak** [comment]
doesn't look like it's installed. try reinstalling it again. amke sure it's sudo

**Comment by monodeepdas1215** [comment]
as you can see from the screenshot above (the first one) it seems I have tried all that could be done to install it. USING sudo

**Comment by ppagan** [comment]
@monodeepdas1215 run `sudo find / -type f -name 'udemy-dl' -exec ls -l {} \;`

That command will find the location of the file 'udemy-dl' anywhere on your system and will do a long-listing on it (ls -l). For example, on my system the output produced:
```
ppagan@nuc:~ > sudo find / -type f -name 'udemy-dl' -exec ls -l {} \; 
[sudo] password for ppagan: 
find: ‘/run/user/1000/gvfs’: Permission denied
-rwxr-xr-x. 1 root root 178 Oct 28 13:05 /root/udemy-dl-0.2.2a2/src/udemy-dl
-rwxr-xr-x. 1 root root 173 Oct 28 13:05 /root/udemy-dl-0.2.2a2/build/scripts-2.7/udemy-dl
-rwxr-xr-x. 1 root root 177 Oct 28 13:05 /root/udemy-dl-0.2.2a2/build/scripts-3.4/udemy-dl
-rwxr-xr-x. 1 root root 175 Oct 29 01:22 /usr/bin/udemy-dl
```
Your output will likely be different, but that file must have been put _somewhere_. Let us know what you find (or don't find).

**Comment by monodeepdas1215** [comment]
![screenshot-2017-10-29 screenshot from 2017-10-29 12-23-32 png png image 1920 x 1080 pixels - scaled 87](https://user-images.githubusercontent.com/15053941/32141354-53d783a2-bca4-11e7-9e2c-e6aa08e8eb33.png)

**Comment by ppagan** [comment]
tldr: You need to update your $PATH variable to contain `$HOME/.local/bin`

$PATH is a semicolon-separated list of directories that tells your system where search for executable files. pip installed udemy-dl in user mode, which put the executable into your ~/.local/bin/ directory, but since that directory isn't in your path variable, your system can't find it when you try to run it. That path variable can be modified for each user by modifying their ~/.bash_profile.

1. Add this line to ~/.bash_profile: `export PATH=$PATH:$HOME/.local/bin`. This is my ~/.bash_profile, as an example:
```
ppagan@nuc:~ > cat ~/.bash_profile 
# .bash_profile

# Get the aliases and functions
if [ -f ~/.bashrc ]; then
        . ~/.bashrc
fi

# User specific environment and startup programs

export PATH=$PATH:$HOME/.local/bin
```

2. Run `source ~/.bash_profile` so your system will pick up the changes.
3. Run `echo $PATH` to make sure that path is in the list.
4. Try again and report back.

**Comment by monodeepdas1215** [comment]
First I tried the commands that were told but this is the output to that
![screenshot-2017-10-30 screenshot from 2017-10-30 15-09-56 png png image 1920 x 1080 pixels - scaled 87](https://user-images.githubusercontent.com/15053941/32155732-350d945a-bd5f-11e7-8dab-f47d7e602838.png)

Then I tried to just export the path to the $PATH variable by
export PATH=$PATH:$HOME/.local/bin
and it seemed something happen when i tested 
udemy-dl

It seemed as if it worked for the first time when it said too few arguments and opened up the help of the command describing the possible parameters but when I restarted the terminal and again typed udemy-dl nothing happened and the same message again
`udemy-dl : command not found`

In the meanwhile when the udemy-dl did show something then I had also tried to download a nodeJS course with it and here is the output to that but once I closed terminal off and restarted terminal the same command not found message ! 
![error](https://user-images.githubusercontent.com/15053941/32155841-f8c7c23a-bd5f-11e7-9a71-3ee1a4055cb7.png)


Please could you tell me how to completely remove each and every bit of the udemy-dl from every part of my OS and then trying to reinstalling it from the beginning. 

I was installing udemy-dl in ubuntu 16.10 LTS and it worked great but I just dont know as if its seems quite different in ubuntu 17.10. I even get errors during installing other packages that the package system is broken. bla bla !
Please suggest some workaround.

**Comment by ppagan** [comment]
When you ran the export command directly from your terminal, it only applied to that terminal session and not any new ones you opened. Also, it looks like Ubuntu uses .bashrc, not .bash_profile (my mistake). Also, every user has their own profile, so setting $PATH for root will not effect the $PATH of your personal user account, and visa versa. Google info on path variables if you can - it's good to know about.

Anyway, the script works, it just looks like you hit another problem (it happens). 

1. Run this from your user account to fix your current path problem:
```
echo 'export PATH=$PATH:$HOME/.local/bin' >> ~/.bashrc
source ~/.bashrc
```

2. You current problem looks like issue #258, which might be solved by running:
```
pip install --user --upgrade pyOpenSSL
```
Try the above, copy/paste the terminal output here (your screenshots are OK but are not text-searchable). Including the flags you entered but do NOT show your password! We can troubleshoot from there.

**Comment by ppagan** [comment]
You hit issue #263. You have your path fixed, and the proper packages installed (I think), so it might just be a version issue. Try against after installing the version mentioned in #263 and see what happens:

`pip install udemy-dl==0.2.2-alpha.2`

**Comment by monodeepdas1215** [comment]
Yes I have already done that and the output is there in the last screenshot.

**Comment by ppagan** [comment]
No, you didn't...not exactly. You installed it using sudo (which means the command ran as root, not as your personal user), and you didn't properly check the version you were running afterwards (`python -m udemy-dl -v` is different from `udemy-dl -v`).

Run `udemy-dl -v` and confirm that you are actually using 'udemy-dl 0.2.2-alpha.2'. If not, carefully rerun the install command (without using sudo), and check the version.

Linux is powerful, but it can be complicated. It's all in the details. Try again and report back.
